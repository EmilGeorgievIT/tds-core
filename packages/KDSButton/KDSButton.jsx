import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components'

import { BaseButton } from '@tds/core-button'

const koodoTheme = {
  generate: (props) => ({
    backgroundColor: 'white',
    color: 'black',
    '&:hover': {
      backgroundColor: 'black',
      color: 'white',
    },
    fontWeight: '900',
    border: '2px solid black',
    borderRadius: '5px',
    padding: '10px'
  })
}

/**
 * @version ./package.json
 */
const KDSButton = (props) => {
  return (
    <ThemeProvider theme={koodoTheme}>
      <BaseButton {...props} />
    </ThemeProvider>
  )
}

KDSButton.propTypes = {
  /**
   * The HTML button type.
   */
  type: PropTypes.oneOf(['button', 'submit', 'reset']),
  /**
   * The style.
   */
  variant: PropTypes.oneOf(['primary', 'secondary']),
}

KDSButton.defaultProps = {
  type: 'button',
  variant: 'primary',
}

export default KDSButton
