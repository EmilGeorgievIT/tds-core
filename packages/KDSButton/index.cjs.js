require('./dist/index.css')
const KDSButton = require('./dist/index.cjs')

module.exports = KDSButton
