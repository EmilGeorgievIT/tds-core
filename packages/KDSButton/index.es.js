import './dist/index.css'
import KDSButton from './dist/index.es'

export default KDSButton
