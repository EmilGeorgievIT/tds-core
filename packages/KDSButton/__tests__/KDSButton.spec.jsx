import React from 'react'
import { shallow, render } from 'enzyme'

import KDSButton from '../KDSButton'

describe('KDSButton', () => {
  const doShallow = (props = {}) => shallow(<KDSButton {...props} />)

  it('renders', () => {
    const kDSButton = render(<KDSButton />)

    expect(kDSButton).toMatchSnapshot()
  })

  it('does other things', () => {
    const kDSButton = doShallow()

    expect(kDSButton).toExist()
  })

  it('passes additional attributes to the element', () => {
    const kDSButton = doShallow({ id: 'the-id', 'data-some-attr': 'some value' })

    expect(kDSButton).toHaveProp('id', 'the-id')
    expect(kDSButton).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const kDSButton = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(kDSButton).not.toHaveProp('className', 'my-custom-class')
    expect(kDSButton).not.toHaveProp('style')
  })
})
