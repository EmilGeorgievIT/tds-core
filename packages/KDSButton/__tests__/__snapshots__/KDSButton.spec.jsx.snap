// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KDSButton renders 1`] = `
.c0 {
  background-color: white;
  color: black;
  font-weight: 900;
  border: 2px solid black;
  border-radius: 5px;
  padding: 10px;
}

.c0:hover {
  background-color: black;
  color: white;
}

<button
  class="c0"
  type="button"
/>
`;
